src/sortStudents.ts

```diff
@@ -1,16 +1,58 @@
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name,
+  Surname,
+  Age,
+  Married,
+  AverageGrade,
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(students: Student[],
+  sortBy: SortType,
+  order: SortOrder): Student[] {
+  const sortedStudents = [...students];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  function getAvarageGrade(grades: number[]): number {
+    const sum = grades.reduce((a, b) => a + b, 0);
+
+    return sum / grades.length;
+  }
+
+  const sortingFunction = (a: Student, b: Student): number => {
+    switch (sortBy) {
+      case SortType.Name:
+        return a.name.localeCompare(b.name);
+      case SortType.Surname:
+        return a.surname.localeCompare(b.surname);
```

Comment:
you can use as a group the same solution for different cases
```suggestion
      case SortType.Name:
      case SortType.Surname:      
        return a[sortBy].localeCompare(b[sortBy]);
```